src/Coordination/KeeperDispatcher.cpp

```diff
@@ -736,6 +732,49 @@ void KeeperDispatcher::updateConfigurationThread()
     }
 }
 
+void KeeperDispatcher::updateConfigurationThread()
+{
+    using namespace std::chrono_literals;
+    while (!shutdown_called)
+    {
+        ConfigUpdateAction action;
+        if (!update_configuration_queue.copyFront(action))
+            return;
+
+        if (server->applyConfigurationUpdate(action))
+        {
+            (void)update_configuration_queue.tryPop(action);
```

Comment:
The idea is, we need to peek the front element of the queue
(as if it's rejected, we don't lose it).
I guess it can also be implemented with pop + pushFront, will reconsider